src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  updateHealth(health) {
+    this.health += health;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive.splice(Animal.alive.indexOf(this), 1);
```

Comment:
[CODE PERFORMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead. This line can be replaced with a filter method to remove the dead animal from the `Animal.alive` array without the need to search for the animal's index first.